include/SANDGeoManager.h

```diff
@@ -136,67 +171,127 @@ class SANDGeoManager : public TObject
+                      const std::map<SANDWireID, SANDWireInfo>& map);
 
  public:
   SANDGeoManager()
       : cellmap_(),
-        sttmap_(),
-        stt_tube_regex_(sand_geometry::stt::stt_tube_regex_string),
-        stt_two_tubes_regex_(sand_geometry::stt::stt_two_tubes_regex_string),
+        stt_tube_regex_(sand_geometry::stt::stt_single_tube_regex_string),
+        // stt_two_tubes_regex_(sand_geometry::stt::stt_two_tubes_regex_string),
         stt_plane_regex_(sand_geometry::stt::stt_plane_regex_string),
         stt_module_regex_(sand_geometry::stt::stt_module_regex_string),
-        stt_tube_tranverse_position_map_()
+        stt_supermodule_regex_(
+            sand_geometry::stt::stt_supermodule_regex_string)
   {
   }
   void init(TGeoManager* const geo);
-  int save_to_file(const char* name = 0, Int_t option = 0, Int_t bufsize = 0)
-  {
-    geo_ = 0;
-    return Write(name, option, bufsize);
-  }
+  void SetGeoCurrentPoint(double x, double y, double z) const;
+  void SetGeoCurrentDirection(double x, double y, double z) const;
+  void InitVolume(volume& v) const;
   const SANDECALCellInfo& get_ecal_cell_info(int ecal_cell_id) const
   {
     return cellmap_.at(ecal_cell_id);
   }
-  const SANDSTTTubeInfo& get_stt_tube_info(int stt_tube_id) const
-  {
-    return sttmap_.at(stt_tube_id);
-  }
+  std::map<SANDTrackerCellID, SANDTrackerCell>::const_iterator get_cell_info(SANDTrackerCellID cell_id) const;
+  plane_iterator get_plane_info(SANDTrackerCellID cell_id) const;
+  plane_iterator get_plane_info(SANDTrackerPlaneID unique_plane_id) const;
   const std::map<int, SANDECALCellInfo>& get_ecal_cell_info() const
   {
     return cellmap_;
   }
-  const std::map<int, SANDSTTTubeInfo>& get_stt_tube_info() const
+  const std::vector<SANDTrackerPlane>&
+      get_planes() const
   {
-    return sttmap_;
+    return _planes;
+  }
+  std::vector<SANDTrackerPlane>&
+      get_planes()
+  {
+    return _planes;
+  }
+
+  const SANDTrackerPlaneIndex GetPlaneIndex(const SANDTrackerPlaneID& plane_uid) const
+  {
+    return SANDTrackerPlaneIndex(std::distance(_planes.cbegin(), _id_to_plane.at(plane_uid)));
   }
   int get_ecal_cell_id(double x, double y, double z) const;
-  int get_stt_tube_id(double x, double y, double z) const;
+  SANDTrackerCellID get_stt_tube_id(double x, double y, double z) const;
+  long print_stt_tube_id(double x, double y, double z) const;
+  
+  // Notice: is the non-const version needed?
+  const TGeoManager* GetTGeoManager() const {return geo_;};
+  TGeoManager* GetTGeoManager() {return geo_;};
+  long get_wire_id(long drift_plane_id, double z,
+                   double transverse_coord) const;
+  std::vector<SANDTrackerCellID> get_segment_ids(const TG4HitSegment& hseg) const;
+  TVector3 FindClosestDrift(TVector3 point, double epsilon) const;
+  TVector3 SmearPoint(TVector3 point, double epsilon) const;
+  
+  const TVector2 pointInRotatedSystem(TVector2 v, double angle) const;
```

Comment:
TVector2 is a legacy class as well.

https://root.cern.ch/doc/master/classTVector2.html